packages/node_modules/@webex/plugin-meetings/src/meeting-info/util.js

```diff
@@ -39,6 +42,43 @@ MeetingInfoUtil.getMeetingLink = (value) => {
   return sipUrl;
 };
 
+// modified from https://github.com/kirm/sip.js base
+// and with https://tools.ietf.org/html/rfc3261
+// requires the @ symbol
+MeetingInfoUtil.isSipUri = (sipString) => {
+  // eslint-disable-next-line no-warning-comments
+  // TODO: lets remove regex from this equation and user URI matchers and such
+  const sipRegex = /^(sips?)?:?(?:([^\s>:@]+)(?::([^\s@>]+))?@)([\w\-.]+)(?::(\d+))?((?:;[^\s=?>;]+(?:=[^\s?;]+)?)*)(?:\?(([^\s&=>]+=[^\s&=>]+)(&[^\s&=>]+=[^\s&=>]+)*))?$/;
+  const sipUri = sipRegex.exec(sipString);
+  return sipUri;
+};
+
+MeetingInfoUtil.generateOptions = (destination, type) => {
+  if (type) {
+    return {
+      destination,
+      type
+    };
+  }
+  const options = {};
+  if (MeetingInfoUtil.isSipUri(destination)) {
+    options.type = SIP_URI;
+    options.destination = destination;
+  }
+  else {
+    const convoUrl = url.parse(destination);
```

Comment:
Is this supporting passing multiple types via `destination` OR specifying the type via `type`? I think we should probably choose one or the other. I preferred the latter but if you and @arun3528 discussed and want to only do the former that's fine. cc @mattnorris

Reply:
right, we can discuss in standup as well here

it supports the following three

pass in conversation url, pass in conversation id, or pass in sip uri, and we don't need to specify type at all for those ones, as that was the request. we had discussed being able to do both convo_url and convo_id in the previous meeting.

We can ALSO specify type via createWithType

Reply:
so if we wanted to change it to specify via type, we save some complexity, but user will have to know the type strings, rather than just knowing convoId or sipUri

to me, being able to do create(destination) where destination is either sipUri, convoId, or convoUrl makes decent sense to me as a shorthand utility, and having createWithType(type, destination) where type and destination are directly mapped makes some sense, but totally open to discussion, i did it based on our talk in the meeting / the comments in the code, but can pretty easily change it

Reply:
I also think once this one gets nailed down i'll write a readme to reflect